dot/telemetry/telemetry.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2021 ChainSafe Systems (ON) Corp.
+// This file is part of gossamer.
+//
+// The gossamer library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The gossamer library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the gossamer library. If not, see <http://www.gnu.org/licenses/>.
+package telemetry
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"sync"
+	"time"
+
+	"github.com/gorilla/websocket"
+	log "github.com/sirupsen/logrus"
+)
+
+// Handler struct for holding telemetry related things
+type Handler struct {
+	buf             bytes.Buffer
+	wsConn          []*websocket.Conn
+	telemetryLogger *log.Entry
+}
+
+// MyJSONFormatter struct for defining JSON Formatter
+type MyJSONFormatter struct {
+}
+
+// Format function for handling JSON formatting, this overrides default logging formatter to remove
+//  log level, line number and timestamp
+func (f *MyJSONFormatter) Format(entry *log.Entry) ([]byte, error) {
+	serialized, err := json.Marshal(entry.Data)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal fields to JSON, %w", err)
+	}
+	return append(serialized, '\n'), nil
+}
+
+var once sync.Once
+var handlerInstance *Handler
+
+// GetInstance singleton pattern to for accessing TelemetryHandler
+func GetInstance() *Handler {
+	if handlerInstance == nil {
+		once.Do(
+			func() {
+				handlerInstance = &Handler{
+					buf: bytes.Buffer{},
+				}
+				log.SetOutput(&handlerInstance.buf)
+				log.SetFormatter(new(MyJSONFormatter))
+			})
+	}
+	return handlerInstance
+}
+
+// AddConnections adds connections to telemetry sever
+func (h *Handler) AddConnections(conns []interface{}) {
+	for _, v := range conns {
+		c, _, err := websocket.DefaultDialer.Dial(fmt.Sprintf("%s", v.([]interface{})[0]), nil)
+		if err != nil {
+			fmt.Printf("Error %v\n", err)
+		}
+		h.wsConn = append(h.wsConn, c)
+	}
+}
+
+// SendConnection sends connection request message to telemetry connection
+func (h *Handler) SendConnection(authority bool, chain, genesis_hash, system_name, node_name,
```

Comment:
Ideally, keep function params size limited to 3. If there are more params group them in a struct.

Reply:
Also, param names should be camel case. `genesis_hash` -> `genesisHash`
Fix in all functions.

Reply:
I created a struct to hold the connection data information.  And changed from snake case to camel case.